app/pages/lab/subject-set.cjsx

```diff
@@ -72,33 +71,53 @@ SubjectSetListing = React.createClass
   getInitialState: ->
     page: 1
     pageCount: NaN
+    subjects: null
 
-  render: ->
+  componentWillReceiveProps: (nextProps) ->
+    @setSubjectResources(nextProps.subjectSet.id, "1")
```

Comment:
This seems a bit odd the page will always be set to 1 when the component receives props. Should there be a conditional here? For example, if I were on page two and selected a subject to upload, the table will jump to page one before I upload the subject.